### ISSUE TITLE
Feature Request JetSmartFilters - Hierarchical Filter: Combination of Radio and Checkbox Filter

### ISSUE THREAD

**Issue by SEwebm** [CrocoBlock/suggestions#6097]
I've tested and researched for plenty of hours, but could not find a method to achieve the following goal:

Let's assume I have two Categories: Cat1 and Cat2.

Furthermore I have three Tags: Tag1, Tag2 and Tag3.

Additionally I have two Posts: Post1 and Post2.

Now to the relations:

Post1 is in Cat1 and has Tag1 and Tag2.
Post2 is in Cat2 and has Tag3.

I want to filter by Categories AND Tags. I am aware that this is possible using either the select hierarchical filter or by a combination of two filters. 
But hear me out first, because those are lacking one crucial point each for my goal.

So what I want exactly is a possibility to not only provide a combined filter, but I want the suggested terms to lead to not empty search results. 

For example if I query for Cat1, I want the second filter - the tag filter - to show only terms that will lead to search results, in this case, _Tag1 and Tag2_. This is possible using the hierarchical select filter, though it is not possible querying for multiple tags, like you can with a combination of radio filter (categories) and checkbox filter (tags). 

Although if you use said combination, then if you query for Cat1: Only Post1 will show up in the search results as it should, but all 3 Tags - _Tag1, Tag2, and Tag3_ will show up as suggested terms in the checkbox filter instead of _Tag 1 and Tag 2_.

So in summary I am missing a feature to exclude terms that in combination with other filters will lead to empty search results OR the possibility to have a combination of radiolike and checkboxlike filters in the hierarchical select filter.

I hope I could break the problem down to the essentials for it to be easily understood. 
Maybe i am missing something and this is already possible, then I would appreciate a hint. If not, maybe someone knows a workaround (hook or even seperate sql query).

Thanks for taking the time to read this.
